### PR TITLE
PLU-261: [onboarding-demo-experiment-5]: add sample template and demo sidebar

### PR DESCRIPTION
**Pull request by m0nggh**
## Problem
New users are confused with how to get started on Plumber, so GGWP v1 is an onboarding demo flow we are experimenting with to provide more help for users to onboard with Plumber.

## Details
Frontend
- Add demo sidebar at the bottom to pop open the `DemoPageModal`
- Loads into the `DemoVideoModalContent` upon proceeding with the demo

## Tests
- [x] Can enter into the demo flow and it will no longer load the tooltip at the flows page

### CODE REVIEW

**Comment by m0nggh** [comment]
* **#622** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/622?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#621** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/621?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#620** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/620?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#619** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/619?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#618** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/618?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#617** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/617?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @m0nggh and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/src/components/Layout/DemoPageModal.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            <Text textStyle="h3-semibold">Send notifications</Text>
+            <Text textStyle="body-1">
+              This demo shows you how to send out a customised email
+              notification with each new form response. Common use cases include
+              notifications, or acknowledgments and follow up instructions.
+            </Text>
+          </Flex>
+
+          <ModalFooter p={0} gap={4}>
+            <Button onClick={onClose} variant="clear" color="primary.600">
```
Comment (pregnantboy) [inline packages/frontend/src/components/Layout/DemoPageModal.tsx:78]: unnecessary lei

File: packages/frontend/src/components/Layout/DemoPageModal.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      variables: {
+        input: {
+          flowName,
+          trigger,
+          actions,
+          demoVideoId,
+        },
+      },
+    })
+    setFlowId(response.data?.createFlowTemplate?.id)
```
Comment (pregnantboy) [inline packages/frontend/src/components/Layout/DemoPageModal.tsx]: why can't u just call navigate here?

File: packages/frontend/src/graphql/mutations/create-flow-template.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { gql } from '@apollo/client'
+
+export const CREATE_FLOW_TEMPLATE = gql`
```
Comment (pregnantboy) [inline packages/frontend/src/graphql/mutations/create-flow-template.ts]: createFlowFromTemplate or createTemplatedFlow

File: packages/frontend/src/components/Layout/DemoPageModal.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+const { flowName, trigger, actions, demoVideoId } =
+  FLOW_TEMPLATES_MAP[FORMSG_POSTMAN_TEMPLATE]
+
+export default function DemoPageModal(props: DemoPageModalProps) {
+  const { onClose } = props
+  const navigate = useNavigate()
+  const [flowId, setFlowId] = useState('')
+
+  const [createFlowTemplate, { loading }] = useMutation(CREATE_FLOW_TEMPLATE)
+  const onCreateFlowTemplate = useCallback(async () => {
```
Comment (pregnantboy) [inline packages/frontend/src/components/Layout/DemoPageModal.tsx]: same issue with naming

File: packages/frontend/src/components/Layout/NavigationSidebar.tsx [hunk 5]
```diff
@@ -56,16 +58,63 @@ function NavigationSidebarItem({
...
+      <SidebarContainer>
+        {links.map((link, index) => (
+          <NavigationSidebarItem
+            key={index}
+            link={link}
+            closeDrawer={closeDrawer}
+          />
+        ))}
+
+        <Box pos="fixed" bottom={0}>
```
Comment (pregnantboy) [inline packages/frontend/src/components/Layout/NavigationSidebar.tsx:115]: dont use position fixed, set margin-top: auto or something

**Review by pregnantboy — APPROVED** [review]
some nits again

File: packages/frontend/src/components/FlowRow/DemoFlowModal.tsx [hunk 6]
```diff
@@ -11,15 +11,26 @@ import {
...
+  DEMO_VIDEOS_MAP,
+  FORMSG_POSTMAN_TEMPLATE,
+} from 'helpers/flow-templates'
 
 import DemoVideoModalContent from './DemoVideoModalContent'
 
 interface DemoFlowModalProps {
   onClose: () => void
+  isPreCreated?: boolean
+  demoVideoId?: string
```
Comment (pregnantboy) [inline packages/frontend/src/components/FlowRow/DemoFlowModal.tsx:24]: demoVideoId = FORMSG_POSTMAN_TEMPLATE

File: packages/frontend/src/components/FlowRow/DemoFlowModal.tsx [hunk 7]
```diff
@@ -11,15 +11,26 @@ import {
...
 }
 
 export default function DemoFlowModal(props: DemoFlowModalProps): ReactElement {
-  const { onClose } = props
+  const { onClose, isPreCreated, demoVideoId } = props
+  // fallback to default demo video to display for GGWP v1
+  const { url, title } = demoVideoId
+    ? DEMO_VIDEOS_MAP[demoVideoId]
+    : DEMO_VIDEOS_MAP[FORMSG_POSTMAN_TEMPLATE]
```
Comment (pregnantboy) [inline packages/frontend/src/components/FlowRow/DemoFlowModal.tsx]: u can either expand the props to { onClose, isPreCreated, demoVideoId = FORMSG_POSTMAN_TEMPLATE }

or

const { url, title } = DEMO_VIDEOS_MAP[demoVideoId ?? FORMSG_POSTMAN_TEMPLATE]

Comment (pregnantboy) [reply packages/frontend/src/components/FlowRow/DemoFlowModal.tsx]: good to also check if the video is exists

File: packages/frontend/src/components/Layout/DemoPageModal.tsx (hunk 2 above)
Comment (m0nggh) [reply packages/frontend/src/components/Layout/DemoPageModal.tsx]: i think i was testing some logic and yes i can do it here thks

File: packages/frontend/src/graphql/mutations/create-flow-template.ts (hunk 3 above)
Comment (m0nggh) [reply packages/frontend/src/graphql/mutations/create-flow-template.ts]: changed to `createTemplatedFlow` thks

File: packages/frontend/src/components/Layout/NavigationSidebar.tsx (hunk 5 above)
Comment (m0nggh) [reply packages/frontend/src/components/Layout/NavigationSidebar.tsx:115]: to fix in another PR (for users who vary the screen size vertically) 😰

**Pull request merged**